### PR TITLE
Change redundant router setting logic while loading APIs

### PR DESCRIPTION
**Pull request by furkansenharputlu**

### CODE REVIEW

**Comment by furkansenharputlu** [comment]
After @gernest's comments, I have done some testing. The logic is not redundant here, it is done by purpose. However, the function and variable names are confusing. Also, a test is missing. All tests are passing although it is a breaking change. The test and variable name changes are necessary in a separate PR.

**Comment by furkansenharputlu** [comment]
This PR is still valid.

File: gateway/api_loader.go [hunk 1]
```diff
@@ -674,18 +674,18 @@ func loadApps(specs []*APISpec) {
...
+	port := globalConf.ListenPort
+
+	if globalConf.ControlAPIPort != 0 {
+		port = globalConf.ControlAPIPort
 	}
+
+	muxer := &proxyMux{}
+	router := mux.NewRouter()
+	loadAPIEndpoints(router)
+	muxer.setRouter(port, "", router)
```
[the remark this answers is not in the document]
Comment (furkansenharputlu) [reply gateway/api_loader.go]: After expanding `TestControlListener`, the tests are still passing. It shows that it is not a breaking change or it doesn't translate logic.

**Comment by buger** [comment]
@gernest your turn :)

**Comment by gernest** [comment]
@furkansenharputlu I see @buger merged your tests into master, can you please rebase now. I need to review with the test in place to make sure we won't miss stuff.

**Comment by furkansenharputlu** [comment]
@gernest Already rebased.

**Pull request merged**